### PR TITLE
Added check for self referencing (#184), check for YList assignment.

### PR DESCRIPTION
**Pull request by psykokwak4**

### CODE REVIEW

File: sdk/python/ydk/services/meta_service.py [hunk 1]
```diff
@@ -220,8 +225,13 @@ def inject_imeta_helper(entity, deviation_table, parent=None):
...
             continue
         elif member.mtype == REFERENCE_LIST:
-            for v in value:
-                inject_imeta_helper(v, deviation_table, entity.i_meta)
+            try:
+                for v in value:
+                    inject_imeta_helper(v, deviation_table, entity.i_meta)
+            except TypeError:
+                raise YPYModelError("Assigned object to YList {},"
+                                    " use list method instead."
```
Comment (abhikeshav) [inline sdk/python/ydk/services/meta_service.py]: Should this error message be ' Assigned object to YList or YLeafList. Use list append() instead.'?

Comment (psykokwak4) [inline sdk/python/ydk/services/meta_service.py]: No, YLeafList has its own reference `REFERENCE_LEAFLIST`.

Comment (abhikeshav) [inline sdk/python/ydk/services/meta_service.py]: OK. I think this should be YPYServiceError. How about the below:

YPYServiceError(' Assigned object to YLeafList. Use list append() instead.'?)

Comment (psykokwak4) [inline sdk/python/ydk/services/meta_service.py]: Sure, will use YPYSerivceError, maybe the error message should be "Assigned object to YList. Use list append or extend method instead." And also, check for YLeafList is in encoder, might need to move here.

**Pull request closed without merging**